draft-davidben-tls-merkle-tree-certs.md

```diff
@@ -475,45 +475,76 @@ The CA also maintains a latest batch number, which is the number of the last bat
 
 For each batch in the "issued" state, the CA maintains the following batch state:
 
-* The list of abridged assertions certified in this batch.
+* The list of entries in the batch, described in {{entries}}.
 
 * The tree head, a hash computed over this list, described in {{building-tree}}.
 
 * A validity window signature computed as described in {{signing}}.
 
 The CA exposes all of this information in an HTTP {{!RFC9110}} interface described in {{publishing}}.
 
+## Batch Entries {#entries}
+
+An issued batch contains a list of batch entries, each of which certifies some assertion up to some expiration time. A batch entry is described by a MerkleTreeEntry structure, defined below.
+
+~~~
+/* In this document, hash is always SHA-256, and hash.length is 32. */
+opaque HashValue[hash.length];
+
+struct {
+    SubjectType subject_type;
+    HashValue subject_info_hash;
+    Claim claims<0..2^16-1>;
+    Timestamp not_after;
+} MerkleTreeEntry;
```

Comment:
Not ideal to have *batch entry* and *merkle tree entry*. We should make this `BatchEntry` or use *Merkle tree entry* everywhere.

Reply:
Ah yeah, done.